api/spec/spec_handler.go

```diff
@@ -847,3 +859,353 @@ func (d *specHandler) SpecFromString(
 	}
 	return true, spec, locator, source, name
 }
+
+func (d *specHandler) RestoreSpecFromOpts(
+	opts map[string]string,
+) (
+	*api.RestoreVolumeSpec,
+	*api.VolumeLocator,
+	error,
+) {
+	spec := &api.RestoreVolumeSpec{}
+	locator := &api.VolumeLocator{}
+	locator.VolumeLabels = make(map[string]string)
+	for k, v := range opts {
+		switch k {
+		case api.SpecNodes:
```

Comment:
This function is mostly a copy paste of UpdateSpecFromOpts. I had to remove a few opt handling since those params are not supported by RestoreVolumeSpec